proof/invariant-abstract/AARCH64/ArchInterrupt_AI.thy

```diff
@@ -109,6 +109,14 @@ lemma (* set_irq_state_valid_cap *)[Interrupt_AI_assms]:
 crunch set_irq_state
   for valid_global_refs[Interrupt_AI_assms]: "valid_global_refs"
 
+lemma deactivateInterrupt_invs:
+  "\<lbrace>invs and (\<lambda>s. interrupt_states s irq \<noteq> IRQInactive) and K config_ARM_GIC_V3\<rbrace>
+   do_machine_op (deactivateInterrupt irq)
+   \<lbrace>\<lambda>rv. invs\<rbrace>"
```

Comment:
Hmm, would `config_ARM_GIC_V3` make more sense as an assumption instead of in the preconditions? It's unlikely to make a difference in practice but having it as an assumption would make a proof fail immediately if a rule with it is used in the wrong configuration, while it being in in the precondition feels to me that it's more likely that it would make it to the final big condition and not be clear where it came from.

Reply:
I initially had it as an explicit assumption, but that does require it to be in the goal immediately, which is not always the case.

Reply:
Ah right, I was thinking that it would come from the config and always be available but I guess that it's more from the guarding ifs.

Reply:
Yes, that’s right. In fact we try to keep the proof agnostic of which config is actually selected, so the only way it should know whether it is on or not should be via explicit case distinction or ‘if’.